Detect tool fails on FreeBSD

The `tools/detect/detect.nim` tool fails on FreeBSD with the following output:

<details>
```
> nim c -r tools/detect/detect.nim
Hint: used config file '/usr/home/euant/src/Nim/config/nim.cfg' [Conf]
Hint: used config file '/usr/home/euant/src/Nim/config/config.nims' [Conf]
..................
/usr/home/euant/src/Nim/tools/detect/detect.nim(50, 3) Hint: 'ifile' is declared but not used [XDeclaredButNotUsed]
CC: detect.nim
Hint:  [Link]
Hint: 44682 lines; 0.861s; 66.336MiB peakmem; Debug build; proj: /usr/home/euant/src/Nim/tools/detect/detect.nim; out: /usr/home/euant/src/Nim/tools/detect/detect [SuccessX]
Hint: /usr/home/euant/src/Nim/tools/detect/detect  [Exec]
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <aio.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <dlfcn.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <errno.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <fcntl.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <fenv.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <fmtmsg.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <fnmatch.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <ftw.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <glob.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <langinfo.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <locale.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <netdb.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <net/if.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <netinet/in.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <netinet/tcp.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <nl_types.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <poll.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <pthread.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sched.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <semaphore.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <signal.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/ipc.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/mman.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/resource.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/select.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/socket.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/stat.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/statvfs.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <sys/wait.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <spawn.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <stdio.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <time.h>
CMD gcc -o testh testh.c
sh: gcc: not found
Not found: <unistd.h>
CMD gcc -o genconsts genconsts.c
sh: gcc: not found
Error: execution of an external program failed: '/usr/home/euant/src/Nim/tools/detect/detect '
```
</details>

Fixing `gcc` to `cc` fixes this, but you then get other errors:

<details>
```
> nim c -r tools/detect/detect.nim
Hint: used config file '/usr/home/euant/src/Nim/config/nim.cfg' [Conf]
Hint: used config file '/usr/home/euant/src/Nim/config/config.nims' [Conf]
..................
/usr/home/euant/src/Nim/tools/detect/detect.nim(50, 3) Hint: 'ifile' is declared but not used [XDeclaredButNotUsed]
CC: detect.nim
Hint:  [Link]
Hint: 44682 lines; 0.860s; 66.336MiB peakmem; Debug build; proj: /usr/home/euant/src/Nim/tools/detect/detect.nim; out: /usr/home/euant/src/Nim/tools/detect/detect [SuccessX]
Hint: /usr/home/euant/src/Nim/tools/detect/detect  [Exec]
CMD cc -o testh testh.c
Found: <aio.h>
CMD cc -o testh testh.c
Found: <dlfcn.h>
CMD cc -o testh testh.c
Found: <errno.h>
CMD cc -o testh testh.c
Found: <fcntl.h>
CMD cc -o testh testh.c
Found: <fenv.h>
CMD cc -o testh testh.c
Found: <fmtmsg.h>
CMD cc -o testh testh.c
Found: <fnmatch.h>
CMD cc -o testh testh.c
Found: <ftw.h>
CMD cc -o testh testh.c
Found: <glob.h>
CMD cc -o testh testh.c
Found: <langinfo.h>
CMD cc -o testh testh.c
Found: <locale.h>
CMD cc -o testh testh.c
Found: <netdb.h>
CMD cc -o testh testh.c
Found: <net/if.h>
CMD cc -o testh testh.c
Found: <netinet/in.h>
CMD cc -o testh testh.c
Found: <netinet/tcp.h>
CMD cc -o testh testh.c
Found: <nl_types.h>
CMD cc -o testh testh.c
Found: <poll.h>
CMD cc -o testh testh.c
Found: <pthread.h>
CMD cc -o testh testh.c
Found: <sched.h>
CMD cc -o testh testh.c
Found: <semaphore.h>
CMD cc -o testh testh.c
Found: <signal.h>
CMD cc -o testh testh.c
Found: <sys/ipc.h>
CMD cc -o testh testh.c
Found: <sys/mman.h>
CMD cc -o testh testh.c
Found: <sys/resource.h>
CMD cc -o testh testh.c
Found: <sys/select.h>
CMD cc -o testh testh.c
Found: <sys/socket.h>
CMD cc -o testh testh.c
Found: <sys/stat.h>
CMD cc -o testh testh.c
Found: <sys/statvfs.h>
CMD cc -o testh testh.c
Found: <sys/wait.h>
CMD cc -o testh testh.c
Found: <spawn.h>
CMD cc -o testh testh.c
Found: <stdio.h>
CMD cc -o testh testh.c
Found: <time.h>
CMD cc -o testh testh.c
Found: <unistd.h>
CMD cc -o genconsts genconsts.c
genconsts.c:472:48: warning: format specifies type 'int' but the argument has type 'const fenv_t *' [-Wformat]
  fprintf(f, "const FE_DFL_ENV* = cint(%d)\n", FE_DFL_ENV);
                                       ~~      ^~~~~~~~~~
/usr/include/fenv.h:99:20: note: expanded from macro 'FE_DFL_ENV'
#define FE_DFL_ENV      (&__fe_dfl_env)
                        ^~~~~~~~~~~~~~~
genconsts.c:1208:58: warning: format specifies type 'int' but the argument has type '__sighandler_t *' (aka 'void (*)(int)') [-Wformat]
  fprintf(f, "const SIG_HOLD* = cast[Sighandler](%d)\n", SIG_HOLD);
                                                 ~~      ^~~~~~~~
/usr/include/sys/signal.h:143:25: note: expanded from macro 'SIG_HOLD'
#define SIG_HOLD        ((__sighandler_t *)3)
                        ^~~~~~~~~~~~~~~~~~~~~
genconsts.c:1211:57: warning: format specifies type 'int' but the argument has type '__sighandler_t *' (aka 'void (*)(int)') [-Wformat]
  fprintf(f, "const SIG_DFL* = cast[Sighandler](%d)\n", SIG_DFL);
                                                ~~      ^~~~~~~
/usr/include/sys/signal.h:139:18: note: expanded from macro 'SIG_DFL'
#define SIG_DFL         ((__sighandler_t *)0)
                        ^~~~~~~~~~~~~~~~~~~~~
genconsts.c:1214:57: warning: format specifies type 'int' but the argument has type '__sighandler_t *' (aka 'void (*)(int)') [-Wformat]
  fprintf(f, "const SIG_ERR* = cast[Sighandler](%d)\n", SIG_ERR);
                                                ~~      ^~~~~~~
/usr/include/sys/signal.h:141:18: note: expanded from macro 'SIG_ERR'
#define SIG_ERR         ((__sighandler_t *)-1)
                        ^~~~~~~~~~~~~~~~~~~~~~
genconsts.c:1217:57: warning: format specifies type 'int' but the argument has type '__sighandler_t *' (aka 'void (*)(int)') [-Wformat]
  fprintf(f, "const SIG_IGN* = cast[Sighandler](%d)\n", SIG_IGN);
                                                ~~      ^~~~~~~
/usr/include/sys/signal.h:140:18: note: expanded from macro 'SIG_IGN'
#define SIG_IGN         ((__sighandler_t *)1)
                        ^~~~~~~~~~~~~~~~~~~~~
genconsts.c:1230:49: warning: format specifies type 'int' but the argument has type 'key_t' (aka 'long') [-Wformat]
  fprintf(f, "const IPC_PRIVATE* = cint(%d)\n", IPC_PRIVATE);
                                        ~~      ^~~~~~~~~~~
                                        %ld
/usr/include/sys/ipc.h:110:21: note: expanded from macro 'IPC_PRIVATE'
#define IPC_PRIVATE     (key_t)0 /* private key */
                        ^~~~~~~~
genconsts.c:1582:54: warning: format specifies type 'long' but the argument has type 'int' [-Wformat]
  fprintf(f, "const CLOCKS_PER_SEC* = clong(%ld)\n", CLOCKS_PER_SEC);
                                            ~~~      ^~~~~~~~~~~~~~
                                            %d
/usr/include/time.h:59:24: note: expanded from macro 'CLOCKS_PER_SEC'
#define CLOCKS_PER_SEC  128
                        ^~~
7 warnings generated.
ld: error: undefined symbol: __fe_dfl_env
>>> referenced by genconsts.c
>>>               /tmp/genconsts-749663.o:(main)
cc: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of an external program failed: '/usr/home/euant/src/Nim/tools/detect/detect '
```
</details>

Related: #12138

--- a/tools/detect/detect.nim
+++ b/tools/detect/detect.nim
@@ -14,7 +14,7 @@
 # compilation.
 import os, strutils
 
-when defined(openbsd):
+when defined(openbsd) or defined(freebsd) or defined(netbsd):
   const
     cc = "cc -o $# $#.c"
     cpp = "cc -E -o $#.i $#.c"
@@ -98,8 +98,8 @@ proc main =
     f.write(nimfile % [other])
     close(f)
 
-  let cCompile = when defined(openbsd): ccLinkMath else: cc
-  let cppCompile = when defined(openbsd): cppLinkMath else: cpp
+  let cCompile = defined(openbsd) or defined(freebsd) or defined(netbsd): ccLinkMath else: cc
+  let cppCompile = defined(openbsd) or defined(freebsd) or defined(netbsd): cppLinkMath else: cpp
   if not myExec(cCompile % [gen.addFileExt(ExeExt), gen]): quit(1)
   if not myExec(cppCompile % [pre.addFileExt(ExeExt), pre]): quit(1)
   when defined(windows):

